src/config.rs

```diff
@@ -82,6 +84,24 @@ impl Config for FsConfig {
         let shell_path = self.root_path.join("shells").join(name);
         shell_path.is_dir()
     }
+
+    fn get_shell_list(&self) -> io::Result<Vec<String>> {
+        let mut shell_names = Vec::new();
+        let root_path = self.shell_root_path().to_owned();
+        if try!(fs::metadata(root_path.to_owned())).is_dir() {
+            for entry in try!(fs::read_dir(root_path)) {
+                let entry = try!(entry);
+                if try!(fs::metadata(entry.path())).is_dir() {
+                    match entry.file_name().into_string() {
+                        Ok(v) => shell_names.push(v),
+                        Err(err) => (),
```

Comment:
usually, if you don't care about the value in a match condition, you can just use `_` as the name.  This prevents warnings about unused variables.